src/core/net/ip6.cpp

```diff
@@ -1068,6 +1069,22 @@ Error Ip6::SendRaw(OwnedPtr<Message> aMessagePtr)
 
     if (header.GetDestination().IsMulticast())
     {
+#if OPENTHREAD_CONFIG_FILTER_LOCAL_ORIGINATED_MULTICAST_PACKETS
+        // For multicast packets originated from this device, only forward to thread
+        // network if destination is in the multicast listener table
+        if (!aMessagePtr->IsOriginThreadNetif() && Get<ThreadNetif>().HasUnicastAddress(header.GetSource()))
+        {
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE && OPENTHREAD_FTD && \
+    OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+            VerifyOrExit((header.GetDestination().IsMulticastLargerThanRealmLocal() &&
+                          Get<BackboneRouter::MulticastListenersTable>().HasListener(header.GetDestination())),
+                         error = kErrorDrop);
+#else
+            ExitNow(error = kErrorDrop);
```

Comment:
So with this, host cannot send a ping (or other messages) to "all-thread nodes" multicast address (`otThreadGetRealmLocalAllThreadNodesMulticastAddress()`) and we would drop it.

Is this the desired behavior?

Reply:
I've updated the filter to only apply to untrusted source on host, OT stack can still send to the "all-thread nodes" address.

I think when untrusted source is sending to multicast scope <= realm-local, then we should filter it so misbehaving applications can't flood the thread network. The application can still communicate with Thread network with subscribed multicast addresses, similar to the other devices on the backbone interface, their packet sent to multicast address <= realm-local scope are not forwarded to Thread network.

Thoughts?

Reply:
@superwhd is also asking same question? https://github.com/openthread/openthread/pull/9735#discussion_r1436012590?

- With this we would not allow `ping` to all (thread) nodes from host inteface?
- I think all messages from Thread interface on a Linux based device will be marked as "untrusted host" origin (so the filter would apply to them).

Reply:
I am okay adding this change. 

Just want to highlight this change in behavior since I know folks may use ping to multicast addresses to test connectivity on Thread, and it will be then dropped by the new filtering mechnasim (may be helpful to add a log to indicate such drops?)

I think the  `cli` ping command which uses `ot::PingSender` will still continue to work multicast address. So that can be used instead?

Reply:
Thanks, updated PR description to note the behavior change with ping command.

Added a log to indicate the drop.